### ISSUE TITLE
CSR for Certificates containing static DH keys

### ISSUE THREAD

**Issue by StefanHri** [cose-wg/CBOR-certificates#80]
EDHOC has a static DH authentication mode. For that mode, we can use CBOR certificates containing the public DH key. According to RFC2986 the CSR contains a signature calculated with the own private key:

```
   The process by which a certification request is constructed involves
   the following steps:

        1. A CertificationRequestInfo value containing a subject
           distinguished name, a subject public key, and optionally a
           set of attributes is constructed by an entity requesting
           certification.

        2. The CertificationRequestInfo value is signed with the subject
           entity's private key.  (See Section 4.2.)

        3. The CertificationRequestInfo value, a signature algorithm
           identifier, and the entity's signature are collected together
           into a CertificationRequest value, defined below.
```
Since we cannot calculate a signature with the private DH key how should we construct a CSR which contains a static public DH key?

**Comment by gselander** [comment]
Proof of possession of static ECDH is defined in section 6 of RFC 6955. This reduces the size of the CSR since the signature is replaced with a MAC. It requires the requesting endpoint to have the static DH key of CA. (This can be acquired using GET /rpks, see draft-selander-ace-coap-est-oscore.)

**Comment by highlunder** [comment]
We belive it is solved through including the needed MAC algorithms in the issuer signature algorithms table

**Comment by gselander** [comment]
> We believe it is solved through including the needed MAC algorithms in the issuer signature algorithms table

... and a note about out of band establishment of public DH key of the verifier. All included in master now, last commit: d818f10.
